src/app/lists/ListsPage.tsx

```diff
@@ -5,60 +5,95 @@ import { Schema } from "@/databaseUtils";
 import supabase from "@/supabaseClient";
 import { fetchComment, fetchLists } from "@/common/fetch";
 import ListsDataView, { ListRow, listsHeaderKeysAndLabels } from "@/app/lists/ListDataview";
+import { logErrorReturnLogId } from "@/logger/logger";
+import { subscriptionStatusRequiresErrorMessage } from "@/common/subscriptionStatusRequiresErrorMessage";
 
 interface FetchedListsData {
     data: Schema["lists"][];
     comment: string;
 }
 
-const fetchData = async (): Promise<FetchedListsData> => {
+const fetchListData = async (): Promise<FetchedListsData> => {
     const [data, comment] = await Promise.all([fetchLists(supabase), fetchComment(supabase)]);
     return { data, comment };
 };
 
+const formatListData = (listData: FetchedListsData): ListRow[] => {
+    return listData.data.map((row) => {
+        return {
+            primaryKey: row.primary_key,
+            rowOrder: row.row_order,
+            itemName: row.item_name,
+            ...Object.fromEntries(
+                listsHeaderKeysAndLabels
+                    .filter(([key]) => /^\d+$/.test(key))
+                    .map(([key]) => [
+                        key,
+                        {
+                            quantity: row[`quantity_for_${key}` as keyof Schema["lists"]],
+                            notes: row[`notes_for_${key}` as keyof Schema["lists"]],
+                        },
+                    ])
+            ),
+        } as ListRow; // this cast is needed here as the type system can't infer what Object.fromEntries will return
+    });
+};
+
 const ListsPage: React.FC<{}> = () => {
     const [isLoading, setIsLoading] = useState(true);
-    const [listsData, setListsData] = useState<ListRow[]>([]);
+    const [listData, setListData] = useState<ListRow[]>([]);
     const [comment, setComment] = useState("");
+    const [errorMessage, setErrorMessage] = useState<string | null>(null);
 
+    function handleSetError(error: string | null): void {
+        error ? setErrorMessage(error) : setErrorMessage(null);
+    }
```

Comment:
Isn't it just this basically? (the child shouldn't be setting the error message to `""` anyway). And if this is the case, we can just pass `setErrorMessage` as a prop instead of defining a whole other function.
```suggestion
    function handleSetError(error: string | null): void {
        setErrorMessage(error)
    }
```